spec/cmab_client_spec.rb

```
@@ -0,0 +1,198 @@
+             .to_return(status: 200, body: {'predictions' => [{'variationId' => 'abc123'}]}.to_json, headers: {'Content-Type' => 'application/json'})
+
+      result = client.fetch_decision(rule_id, user_id, attributes, cmab_uuid)
+
+      expect(result).to eq('abc123')
+      expect(WebMock).to have_requested(:post, expected_url)
+                     .with(body: expected_body_for_webmock, headers: expected_headers).once
+      expect(Kernel).not_to have_received(:sleep)
+    end
+
+    it 'should return HTTP exception' do
+      WebMock.stub_request(:post, expected_url)
+             .with(body: expected_body_for_webmock, headers: expected_headers)
+             .to_raise(StandardError.new('Connection error'))
+
+      expect do
+        client.fetch_decision(rule_id, user_id, attributes, cmab_uuid)
+      end.to raise_error(Optimizely::CmabFetchError, /Connection error/)
+
+      expect(WebMock).to have_requested(:post, expected_url)
+                     .with(body: expected_body_for_webmock, headers: expected_headers).once
+      expect(spy_logger).to have_received(:log).with(Logger::ERROR, a_string_including('Connection error'))
+      expect(Kernel).not_to have_received(:sleep)
+    end
+
+    it 'should not return 200 status' do
+      WebMock.stub_request(:post, expected_url)
+             .with(body: expected_body_for_webmock, headers: expected_headers)
+             .to_return(status: 500)
+
+      expect do
+        client.fetch_decision(rule_id, user_id, attributes, cmab_uuid)
+      end.to raise_error(Optimizely::CmabFetchError, /500/)
+
+      expect(WebMock).to have_requested(:post, expected_url)
+                     .with(body: expected_body_for_webmock, headers: expected_headers).once
+      expect(spy_logger).to have_received(:log).with(Logger::ERROR, a_string_including('500'))
+      expect(Kernel).not_to have_received(:sleep)
+    end
+
+    it 'should return invalid json' do
```

how about `should raise error on invalid json response`?